friends/src/components/Form.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react'
+// import axios from 'axios':
+// import PropTypes from 'prop-types'
+
+class Form extends Component {
+  constructor(props) {
+    super(props)
+    this.state ={
+      name:'',
+      age: '',
+      email: '',
+      currentFriend:{}
+    }
+  }
+
+  ComponentDidMount(){
+    let currentFriend = this.props.data.find( item=> item.id.tostring() === this. match.params.id);
+  };
+
+  // this.setState({
+  //   name: this.props.edit ? currentFriend.name :  "",
+  // })
+
+  // componentDidUpdate(prevProps, prevState) {
+
+  //   if (prevProps.currentFriend.name != this.props.currentFriend.name) {
+
+  //     this.setState({
+  //       name: this.props.currentFriend.name,
+  //       age: this.props.currentFriend.age,
+  //       email: this.props.currentFriend.email,
+  //     })
+
+  //   }
+
+  // }
+    
+  onChangeHandler = event =>{
+    event.preventDefault();
+      this.setState({
+        [event.target.name]: event.target.value
+      });
+    };
+
+  submitHandler = event => { 
+    event.preventDefault()
+    this.props.onAddFriend(this.state);
+    this.props.history.push("/Friends");
+  };
+
+  render() {
+    return (
+      <div>
+      <h1>{this.props.onUpdate ? "Update Friend" : "Add New Friend"}</h1>
+      <form onSubmit={event => {
+        event.preventDefault ();
+        this.props.addFriend(this.state);this.props.history.push ('/Friends');}}>
```

Comment:
Why have this all on the same line? Anything separated by semicolons usually belong on separate lines for the sake of readability.